src/functional/iterator/embedded.py

```diff
@@ -483,6 +483,12 @@ def promote_scalars(val: CompositeOfScalarOrField):
     if math_builtin_name == "gamma":
         # numpy has no gamma function
         impl = np.vectorize(math.gamma)
+    elif math_builtin_name in ["int", "float", "bool", "str"]:
+        # TODO: Using python builtins instead of numpy fixed size types to keep
+        # behaviour the same. Maybe change in the future.
+        import builtins as python_builtins  # noqa: T100 functional.iterator.builtins shadows builtins
```

Comment:
you already import builtins on line 34

Reply:
That's precisely why this is needed. I have to access the Python `builtins`, not `functional.iterator.builtins`, which is being imported there.

Reply:
But I guess it's also possible to import `functional.iterator.builtins` as an alias, or add 4 special cases instead of this blanket one.

Reply:
The solution you picked in the end is fine I think. In principle the `functional.iterator.builtins` should not have been called `builtins` to avoid the shadowing you stumbled upon here.